apps/passport-server/src/services/telegramService.ts

```diff
@@ -819,6 +811,44 @@ export class TelegramService {
     );
   }
 
+  public async handleRequestAnonymousMessageLink(
+    telegramChatId: number,
+    topicId: number
+  ): Promise<string> {
+    // Confirm that topicId exists and is anonymous
+    const topics = await fetchTelegramAnonTopicsByChatId(
+      this.context.dbPool,
+      telegramChatId
+    );
+    const topic = topics.find(
+      (t) => t.topic_id === topicId.toString() && t.is_anon_topic
+    );
```

Comment:
make this an sql query

Reply:
Noted here https://github.com/proofcarryingdata/zupass/issues/901